scene/resources/texture.cpp

```diff
@@ -915,7 +915,11 @@ uint32_t AtlasTexture::get_flags() const {
 }
 
 void AtlasTexture::set_atlas(const Ref<Texture> &p_atlas) {
-	ERR_FAIL_COND(p_atlas == this);
+	Ref<AtlasTexture> at = p_atlas;
+	while (at.is_valid()) {
+		ERR_FAIL_COND_MSG(at == this, vformat("Can't set %s as atlas of %s as it would result in a cyclic dependency.", p_atlas, this));
```

Comment:
Interesting, is there an implicit conversion from `Ref<T>` to `String` for use in `vformat`? I used to call `->to_string()` manually.

Reply:
Actually I haven't even thought of that, just wrote it like this and it worked. :smile: 

Just looked at it and `vformat` seems to take in `const Variant &` and there's a `operator Variant()` in `Ref<T>`. And `vformat` passes such `Variant` to `String::sprintf` which, in case of `%s`, (indirectly) calls `Variant::operator String()` which calls `Variant::stringify`. Something like that.